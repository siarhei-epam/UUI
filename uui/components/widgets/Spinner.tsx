import * as React from 'react';
import { Spinner as uuiSpinner, SpinnerProps as UuiSpinnerProps } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import css from './Spinner.scss';

export type SpinnerProps = UuiSpinnerProps;

export function applySpinnerMods() {
    return [
        css.root,
    ];
}

export const Spinner = withMods<UuiSpinnerProps>(uuiSpinner, applySpinnerMods);
