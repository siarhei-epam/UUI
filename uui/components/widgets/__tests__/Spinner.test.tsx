import React from 'react';
import { Spinner } from '../Spinner';
import renderer from 'react-test-renderer';

describe('Spinner', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<Spinner />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

