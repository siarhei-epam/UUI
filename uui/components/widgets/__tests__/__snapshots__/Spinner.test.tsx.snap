// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should be rendered correctly 1`] = `
<div
  className="container uui-spinner-container root"
>
  <div
    className="uui-spinner-animation"
  >
    <div
      className="uui-spinner-dot uui-spinner-dot-1"
    />
    <div
      className="uui-spinner-dot uui-spinner-dot-2"
    />
    <div
      className="uui-spinner-dot uui-spinner-dot-3"
    />
  </div>
</div>
`;
